services/blockchain-indexer/shared/dataService/business/interoperability/chainAccount.js

```diff
@@ -0,0 +1,26 @@
+/*
+ * LiskHQ/lisk-service
+ * Copyright © 2023 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+
+const { requestConnector } = require('../../../utils/request');
+
+const getChainAccount = async params => {
```

Comment:
May be we can completely remove the method `getChainAccount` from `dataService` layer as it is not been used in `dataService` and we can directly request the connector in indexer wherever it needed. Just a suggestion.

Reply:
It's fine either ways. It probably will be cleaner to directly invoke the connector with `requestConnector` in the `getChainStatus` method in `services/blockchain-indexer/shared/indexer/transactionProcessor/interoperability/registerMainchain.js`.

`Ref: services/blockchain-indexer/shared/indexer/transactionProcessor/index.js`

Reply:
I can understand that we are passing the `getChainAccount` through layers which are not using it. But doing this is keeping all the module related invocations in one place and also keeping (currently most of the) the external calls  under one directory path. For this reason, I would prefer exposing it from the dataservice. Please correct me incase I am headed to a wrong direction guys.